### PR TITLE
Add changelog entry for ppxlib.0.31.0 release

### PR DESCRIPTION
**Pull request by pitag-ha**
I've just opened the opam-repo PR for the ppxlib.0.31.0 release https://github.com/ocaml/opam-repository/pull/24497.

Here's how I'd summarize the release in the ocaml.org changelog. Does that sound good?

I'll undraft this PR as soon as the opam-repo PR is merged.

### CODE REVIEW

File: data/changelog/ppxlib/2023-09-21-ppxlib-0.31.0.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - Preserve quoted attributes on antiquotes in metaquot. (#441, @ncik-roberts)
+
+    - Attribute namespaces: Fix semantics of reserving multi-component namespaces. (#443, @ncik-roberts)
+---
+
+
+We're happy to announce the release of Ppxlib 0.31.0. Shortly after OCaml 5.1.0 has been released,
+this Ppxlib release fixes a bug in the support of OCaml 5.1.0. Before that bug fix, the warnings
+introduced by OCaml 5.1.0 about a generative/applicative missmatch between the creation and application
+of a functor were also triggered when that mismatch didn't exist.
```
Comment (christinerose) [inline data/changelog/ppxlib/2023-09-21-ppxlib-0.31.0.md]:
```suggestion
We're happy to announce the release of `Ppxlib` 0.31.0. Shortly after OCaml 5.1.0 has been released,
this `Ppxlib` release fixes a bug in the support of OCaml 5.1.0. Before that bug fix, the warnings about a generative/applicative mismatch between the creation and functor application introduced by OCaml 5.1.0 were also triggered when that mismatch didn't exist.
```

File: data/changelog/ppxlib/2023-09-21-ppxlib-0.31.0.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Furthermore, the release contains a couple of bug fixes in the context of attributes.
+
+We're also excited about two main enhancements. One allows authors of extension node rewriters to add a
+path argument to the extension node. That's excellent for hygiene since it allows the PPX to be explicit
+about modules rather than depending on its scope.
+
+The other main enhancement allows to opt in for compiler warnings about unused code generated by
```
Comment (christinerose) [inline data/changelog/ppxlib/2023-09-21-ppxlib-0.31.0.md]:
```suggestion
The other main enhancement allows an opt-in for compiler warnings about unused code generated by
```

**Review by christinerose — COMMENTED** [review]
Syntax suggestions and minor formatting suggestion

**Comment by pitag-ha** [comment]
Thanks @christinerose! I've committed both suggestions.

**Review by christinerose — APPROVED** [review]

**Comment by pitag-ha** [comment]
The opam-repo release is through now: https://github.com/ocaml/opam-repository/pull/24497 has been merged. So this PR can be merged, I've just undrafted it. Thanks!

**Comment by tmattio** [comment]
Thank you!

**Pull request merged**